json.Marshal() and sheriff.Marshal() gives different results when empty map present in struct

When empty map is present in Structure sheriff.Marshal() gives `nil` but json.Marshal() gives empty json `{}`

I used existing test `TestMarshal_EmptyMap` and Marshal same object using json & sheriff and got different results.

Modified Test func:
```
func TestMarshal_EmptyMap(t *testing.T) {                   
        emp := EmptyMapTest{                                
                AMap: make(map[string]string),              
        }                                                   
        o := &Options{                                      
                Groups: []string{"test"},                   
        }                                                   
                                                            
        actualMap, err := Marshal(o, emp)                   
        assert.NoError(t, err)                              
                                                            
        actual, err := json.Marshal(actualMap)              
        assert.NoError(t, err)                              
                                                            
        expected, err := json.Marshal(emp)                  
        assert.NoError(t, err)                              
                                                            
        assert.Equal(t, string(expected), string(actual))   
}
``` 

Test result:
```
--- FAIL: TestMarshal_EmptyMap (0.00s)
    sheriff_test.go:510:
                Error Trace:    sheriff_test.go:510
                Error:          Not equal:
                                expected: "{\"a_map\":{}}"
                                actual  : "{\"a_map\":null}"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -{"a_map":{}}
                                +{"a_map":null}
                Test:           TestMarshal_EmptyMap
FAIL
```

--- a/sheriff.go
+++ b/sheriff.go
@@ -248,7 +248,7 @@ func marshalValue(options *Options, v reflect.Value) (interface{}, error) {
 	if k == reflect.Map {
 		mapKeys := v.MapKeys()
 		if len(mapKeys) == 0 {
-			return nil, nil
+			return val, nil
 		}
 		if mapKeys[0].Kind() != reflect.String {
 			return nil, MarshalInvalidTypeError{t: mapKeys[0].Kind(), data: val}

--- a/sheriff_test.go
+++ b/sheriff_test.go
@@ -505,8 +505,28 @@ func TestMarshal_EmptyMap(t *testing.T) {
 	assert.NoError(t, err)
 
 	expected, err := json.Marshal(map[string]interface{}{
-		"a_map": nil,
+		"a_map": make(map[string]interface{}),
 	})
+	assert.NoError(t, err)
+
+	assert.Equal(t, string(expected), string(actual))
+}
+
+func TestMarshal_EmptyMapJson(t *testing.T) {
+	emp := EmptyMapTest{
+		AMap: make(map[string]string),
+	}
+	o := &Options{
+		Groups: []string{"test"},
+	}
+
+	actualMap, err := Marshal(o, emp)
+	assert.NoError(t, err)
+
+	actual, err := json.Marshal(actualMap)
+	assert.NoError(t, err)
+
+	expected, err := json.Marshal(emp)
 	assert.NoError(t, err)
 
 	assert.Equal(t, string(expected), string(actual))

